ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCaseSpecNode.java

```diff
@@ -0,0 +1,49 @@
+package io.confluent.ksql.test.planned;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.ksql.test.model.RecordNode;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class TestCaseSpecNode {
+  private final String version;
+  private final long timestamp;
+  private final Map<String, String> schemas;
+  private final List<RecordNode> inputs;
+  private final List<RecordNode> outputs;
+
+  public TestCaseSpecNode(
+      @JsonProperty("version") final String version,
+      @JsonProperty("timestamp") final long timestamp,
+      @JsonProperty("schemas") final Map<String, String> schemas,
+      @JsonProperty("inputs") final List<RecordNode> inputs,
+      @JsonProperty("outputs") final List<RecordNode> outputs
+  ) {
+    this.version = Objects.requireNonNull(version, "version");
+    this.timestamp = timestamp;
+    this.schemas = Objects.requireNonNull(schemas, "schemas");
+    this.inputs = inputs; // Objects.requireNonNull(inputs, "inputs");
+    this.outputs = outputs; // Objects.requireNonNull(outputs, "outputs");
```

Comment:
nit: commented out code.

If these can be null I'd recommend making them `Optional`.  (You can plug Java8 support into Jackson if the mapper you're using doesn't have this already).

Plus, i'd recommend taking an immutable copy of the input.